### DISCUSSION TITLE
Can I just parse a dotenv file to a HashMap<String, String>?

### DISCUSSION THREAD

**Discussion by alshdavid** [allan2/dotenvy#105] [General]
Looking at the API it seems to load the dotenv file into the process environment. This is undesirable for my use case as I'd like to simply parse the env file from a specified file path to a `HashMap<String, String>`.

Is this possible with the current crate?

**Comment by allan2** [comment]
@alshdavid, this is precisely the new recommended behaviour in [v0.16](https://github.com/allan2/dotenvy/blob/fa19b77fa20b139ddc6a0d5fc12cfabd87d637d7/dotenvy/src/lib.rs).

`EnvLoader::from_path(path).load()` returns `HashMap<String, String>` and does not modify the environment.
For those who do wish to modify the environment, there is the unsafe `EnvLoader::load_and_modify`.
